### PR TITLE
Add my part of code and tests at introduction to classes

### PR DESCRIPTION
**Pull request by nicolas13sochi**
Dmitry, please, take a look... I've checked it before push and it was nice...

### CODE REVIEW

File: codecademy/introduction_to_classes.py [hunk 1]
```diff
@@ -10,15 +10,16 @@ def __init__(self, name, color, flavor, poisonous, weight):
     def description(self):
         return "I'm a %s %s and I taste %s." % (self.color, self.name, self.flavor)
 
-    def is_edible(self, type_posinous):
-        assert type(type_posinous) == bool, "Error type, please enter boolean"
+    def is_edible(self, type_poisinous):
+        assert type(type_poisinous) == bool, "Error type, please enter boolean"
```
Comment (dmitrytokarev) [inline codecademy/introduction_to_classes.py]: poisonous

File: codecademy/introduction_to_classes.py [hunk 2]
```diff
@@ -10,15 +10,16 @@ def __init__(self, name, color, flavor, poisonous, weight):
...
         if not self.poisonous:
             return True
         else:
             return False
 
     def bite_it(self, bite_size):
-        assert type(bite_size) in (int, float) and bite_size > 0, "{} error type, please enter integer".format(bite_size)
+        assert type(bite_size) in (int, float) and bite_size > 0,\
+            "{} error type, please enter integer".format(bite_size)
```
Comment (dmitrytokarev) [inline codecademy/introduction_to_classes.py]: no need to make short lines.
We will be OK with long lines if that doesn't really affect readability

File: codecademy/introduction_to_classes.py [hunk 3]
```diff
@@ -67,12 +69,126 @@ def __init__(self, angle1, angle2, angle3):
...
+        for a in (angle1, angle2, angle3):
+            assert type(a) in (int, float), "Error type, please enter a valid value"
+            assert 0 < a < 180, "Angles is not in the range (0, 180)"
+        assert self.check_angles12(), "Sum of given angles not equal to 180"
+
+    def check_angles12(self):
+        if (self.angle1 + self.angle2 + self.angle3) == 180:
+            return True
+        else:
+            return False
```
Comment (dmitrytokarev) [inline codecademy/introduction_to_classes.py]: no need to do if-else here.
just return the boolean statement:
```python
return self.angle1 + self.angle2 + self.angle3 == 180
```

File: tests/test_introduction_to_classes.py [hunk 4]
```diff
@@ -5,6 +5,11 @@
 from codecademy.introduction_to_classes import Fruit
 from codecademy.introduction_to_classes import ReturningCustomer
 from codecademy.introduction_to_classes import Triangle
+from codecademy.introduction_to_classes import Animal6
+from codecademy.introduction_to_classes import ShoppingCart10
+from codecademy.introduction_to_classes import Shape12, Triangle12, Equilateral12
+from codecademy.introduction_to_classes import Employee14, PartTimeEmployee14
```
Comment (dmitrytokarev) [inline tests/test_introduction_to_classes.py]: please chain imports from same module

File: tests/test_introduction_to_classes.py [hunk 5]
```diff
@@ -248,4 +253,319 @@ def test_check_angles_sum_neg(test_angle1, test_angle2, test_angle3):
...
+    6
+])
+@pytest.mark.parametrize("test_animal_is_hungry", [
+    None,
+    "Yes",
+    "No",
+    0,
+    -1,
+    ""
+])
```
Comment (dmitrytokarev) [inline tests/test_introduction_to_classes.py]: It'd be more readable if your list doesn't contain data structures (lists, tuples, dicts) and you put all values in 1 line:
```python
@pytest.mark.parametrize("test_animal_name_negative", [ None, True, False, 0, "Joffrey" ])
@pytest.mark.parametrize("test_animal_age_negative", [ None, True, False, 0, -1, "", 6 ])
@pytest.mark.parametrize("test_animal_is_hungry", [ None, "Yes", "No", 0, -1, "" ])
```

Comment (dmitrytokarev) [reply tests/test_introduction_to_classes.py]: please do this for other newly written tests.

File: tests/test_introduction_to_classes.py (hunk 4 above)
Comment (nicolas13sochi) [reply tests/test_introduction_to_classes.py]: Dima, I do not understand what I did wrong?

File: tests/test_introduction_to_classes.py (hunk 5 above)
Comment (nicolas13sochi) [reply tests/test_introduction_to_classes.py]: Should I do it for the all same situations in code? (I mean Sanya's part and my one)

Comment (dmitrytokarev) [reply tests/test_introduction_to_classes.py]: Well just for the new tests, readability counts!

File: tests/test_introduction_to_classes.py (hunk 4 above)
Comment (dmitrytokarev) [reply tests/test_introduction_to_classes.py]: Nothing really wrong, just chain your imports  == put imorts from same module on 1 line, with comma as a delimiter

Comment (dmitrytokarev) [reply tests/test_introduction_to_classes.py]: and sort the imported classes, functions alphabetically

File: tests/test_introduction_to_classes.py [hunk 6]
```diff
@@ -1,14 +1,14 @@
...
+from codecademy.introduction_to_classes import Employee14, PartTimeEmployee14
 from codecademy.introduction_to_classes import Fruit
 from codecademy.introduction_to_classes import ReturningCustomer
-from codecademy.introduction_to_classes import Triangle
-from codecademy.introduction_to_classes import Animal6
-from codecademy.introduction_to_classes import ShoppingCart10
 from codecademy.introduction_to_classes import Shape12, Triangle12, Equilateral12
-from codecademy.introduction_to_classes import Employee14, PartTimeEmployee14
+from codecademy.introduction_to_classes import ShoppingCart10
+from codecademy.introduction_to_classes import Triangle
```
Comment (dmitrytokarev) [inline tests/test_introduction_to_classes.py]: @nicolas13sochi chain imports is this:
```python
from codecademy.introduction_to_classes import Animal, Animal6, Customer, Employee14, PartTimeEmployee14, Fruit, ReturningCustomer, Shape12, Triangle12, Equilateral12, ShoppingCart10, Triangle
```
But you need to sort them alphabetically

File: tests/test_introduction_to_classes.py [hunk 7]
```diff
@@ -390,37 +349,15 @@ def test_Triangle12(test_angle1, test_angle2, test_angle3):
...
-    -1,
-    180,
-    0,
-    True,
-    None,
-])
+@pytest.mark.parametrize("test_angle1", [False, -1, 180, 0, True, None])
+@pytest.mark.parametrize("test_angle2", [False, -1, 180, 0, True, None])
+@pytest.mark.parametrize("test_angle3", [False, -1, 180, 0, True, None])
 def test_Triangle12_negative(test_angle1, test_angle2, test_angle3):
```
Comment (dmitrytokarev) [inline tests/test_introduction_to_classes.py]: nice and clean!!! 👍

File: tests/test_introduction_to_classes.py [hunk 8]
```diff
@@ -436,31 +373,9 @@ def test_check_angles12(test_angle1, test_angle2, test_angle3, expected):
...
-    179,
-    61,
-    179,
-    1,
-    20,
-    55,
-])
+@pytest.mark.parametrize("test_angle1", [1, 60, 179, 1, 20, 55])
+@pytest.mark.parametrize("test_angle2", [1, 61, 179, 1, 20, 55])
+@pytest.mark.parametrize("test_angle3", [179, 61, 179, 1, 20, 55])
```
Comment (dmitrytokarev) [inline tests/test_introduction_to_classes.py]: 👍

**Pull request merged**